olive/passes/onnx/optimum_merging.py

```diff
@@ -61,12 +69,13 @@ def new_byte_size_func(_):
 
         olive_model = model_proto_to_olive_model(merged_model, output_model_path, config)
 
-        # Doing a dry run of ORT allows us to remove the initializers that were orphaned by the merging step
-        sess_options = onnxruntime.SessionOptions()
-        sess_options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_ENABLE_BASIC
-        sess_options.optimized_model_filepath = output_model_path
+        if config["run_ort_postprocess"]:
+            # Doing a dry run of ORT allows us to remove the initializers that were orphaned by the merging step
+            sess_options = onnxruntime.SessionOptions()
+            sess_options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_ENABLE_BASIC
+            sess_options.optimized_model_filepath = output_model_path
 
-        execution_provider = self.accelerator_spec.execution_provider
-        onnxruntime.InferenceSession(output_model_path, sess_options, providers=[execution_provider])
+            execution_provider = self.accelerator_spec.execution_provider
+            onnxruntime.InferenceSession(output_model_path, sess_options, providers=[execution_provider])
```

Comment:
would the write file fail since the input model and optimized_model_filepath point to the same value?

Reply:
we might need add an unit test to cover this case. I am supposing it would fail

Reply:
It doesn't fail, it's the default behavior since the beginning. It's able to write to the same location as the input (once the ONNX model is loaded in memory it doesn't keep a handle on it, so it's able to save to the same location).

Reply:
I asked a similar question before but wanted to confirm with the external data. Do you know what happens to those? 
Does the graph optimization know what the original locations of the external data were? or they don't matter since it only cares about the .onnx file and the graph?

Reply:
I don't think it cares about where the weights are located (since those shouldn't change), but the .onnx file has a reference to the location of external data anyway so it **should** be able to find it if needed.

Reply:
I am not sure whether the path is locked or not in Windows by considering the bug fix https://github.com/microsoft/onnxruntime/pull/17697. It would be better we can add unit test to cover it.